pkg/nimble/rpble/include/nimble_rpble.h

```diff
@@ -0,0 +1,178 @@
+ * node starts to advertise its own RPL context data to accept connections
+ * from potential child nodes.
+ *
+ * This approach leads to a BLE network topology that is equal to the IP routing
+ * topology created by RPL on top.
+ *
+ * ## Advertising data structure
+ * To include RPL context information into (legacy) BLE advertisements, it must
+ * be encoded into the BLE advertising data format. This implementation uses
+ * a custom sub-format that is included into the 16-bit UUID Service Data
+ * (type: 0x16) field (Supplement to Bluetooth Core Specification CSSv8, 1.11).
+ * The 16-bit UUID in this field is set to the IPSS service ID (0x1820).
+ *
+ * The following sub-format is used to encode the RPL context data:
+ * ```
+ * byte     1: instance ID (1b)
+ * byte  2-17: DODAG ID (16b)
+ * byte    18: DODAG version (1b)
+ * byte    19: RPL role (1b)
+ * byte 20-21: rank (2b)
+ * byte    22: number of free BLE connection slots (1b)
+ * ```
+ *
+ * ## Ranking of potential parents
+ * The currently implemented ranking metric is very simple: the potential parent
+ * node with the smallest rank is chosen. Additionally, nodes advertise the
+ * number of free BLE connection slots. In case where multiple nodes advertise
+ * the same RPL rank, the one with the largest number of open BLE connection
+ * slots is selected. The idea behind this is to balance the number of BLE
+ * connections per node, and with that also to balance the RPL DODAG.
+ *
+ *
+ * # Usage
+ * To use this module, simply include `nimble_rpble` into your build. If nothing
+ * is explicitly configured, the module will use the default configuration as
+ * specified in `pkg/nimble/rpble/include/nimble_rpble_params.h`.
+ *
+ * Once a node is configured as RPL root (e.g. using the `rpl root ..` shell
+ * command), it will automatically start to advertise itself. Non-RPL-root nodes
+ * will automatically scan for potential parent nodes and join the network as
+ * soon as they find fitting neighbors.
+ *
+ *
+ * ## Implementation status
+ * In its current state, the implementation only works for environments where a
+ * single RPL network with a single DODAG are present. The DODAG ID, instance
+ * ID, and DODAG version are currently pretty much ignored when scanning for
+ * potential parents.
+ *
+ * @{
+ *
+ * @file
+ * @brief       Interface for the nimble_rpble module
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ */
+
+#ifndef NIMBLE_RPBLE_H
+#define NIMBLE_RPBLE_H
+
+#include "nimble_netif.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   rpble configuration parameters
+ */
+typedef struct {
+    uint32_t scan_itvl;         /**< scan interval when scanning for parents */
+    uint32_t scan_win;          /**< scan window when scanning for parents */
+    uint32_t adv_itvl;          /**< advertising interval used when advertising
+                                 *   RPL context to child nodes */
```

Comment:
what units are those? ms? µs?
would 16 bit be enough here?

Reply:
ms, will add better doc. 16-bit is enough (its inherited from the NimBLE APIs)